app/controllers/vm_common.rb

```diff
@@ -1186,11 +1214,28 @@ def open_parent_nodes(record)
     add_nodes
   end
 
+  # if node is VM or Template and hide_vms is true - select parent node in explorer tree but show info of Vm/Template
+  def resolve_node_info(id)
+    nodetype, id = id.split("-")
+
+    if hide_vms && (nodetype == 'v' || nodetype == 't')
+      self.x_node = parent_folder_id(VmOrTemplate.find(id))
+      @vm = VmOrTemplate.find(id)
+    else
+      self.x_node = "#{nodetype}-#{to_cid(id)}"
+    end
+    get_node_info("#{nodetype}-#{to_cid(id)}")
+  end
+
   # Get all info for the node about to be displayed
   def get_node_info(treenodeid)
     # resetting action that was stored during edit to determine what is being edited
     @sb[:action] = nil
-    @nodetype, id = parse_nodetype_and_id(valid_active_node(treenodeid))
+    @nodetype, id = if @vm.present? && hide_vms
```

Comment:
This `@vm.present? && hide_vms` bit is copied over and over.  Would it be possible to extract that out, or somehow have hide_vms just encapsulate the `@vm.present?`  What does `@vm.present?` even mean, code-wise.  Does that mean a Vm is currently selected?  If so, it might make the code clearer to say `if vm_selected` or something like that.